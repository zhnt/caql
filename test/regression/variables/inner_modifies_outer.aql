// Inner block modifies outer variable
let x = 10
let y = 20
print("before: x=" + x + ", y=" + y)
{
    x = 100  // modify outer variable
    let z = 30
    y = y + z  // modify outer variable using local
    print("inside: x=" + x + ", y=" + y + ", z=" + z)
}
print("after: x=" + x + ", y=" + y)

