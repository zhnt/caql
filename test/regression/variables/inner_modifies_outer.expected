before: x=10, y=20
inside: x=100, y=50, z=30
after: x=100, y=50

